File: web/src/types/gate.ts
export interface Gate {
  id: string;
  gate_type: string;
  task_id: string;
  agent_role: string;
  content: Record<string, string> | null;
  status: 'pending' | 'approved' | 'rejected' | 'revised';
  reviewer: string | null;
  review_comment: string | null;
  created_at: string;
  reviewed_at: string | null;
  // Phase 1.3: Retry tracking
  retry_count?: number;
  // Phase 2.3: Dynamic gate flag
  is_dynamic?: boolean;
  // Phase 2.4: Revised content
  revised_content?: string | null;
}

export interface GateApproveRequest {
  reviewer?: string;
  comment?: string;
}

export interface GateRejectRequest {
  reviewer?: string;
  comment: string;
}

export interface GateReviseRequest {
  reviewer?: string;
  comment: string;
  revised_content?: string;
}

File: platform/app/schemas/gate.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class GateDetailResponse(BaseModel):
    id: str
    gate_type: str
    task_id: str
    agent_role: str
    content: Optional[dict] = None
    status: str
    reviewer: Optional[str] = None
    review_comment: Optional[str] = None
    created_at: datetime
    reviewed_at: Optional[datetime] = None
    # Phase 1.3: Retry tracking
    retry_count: int = 0
    # Phase 2.3: Dynamic gate flag
    is_dynamic: bool = False
    # Phase 2.4: Revised content
    revised_content: Optional[str] = None

    model_config = {"from_attributes": True}


class GateListResponse(BaseModel):
    items: List[GateDetailResponse]
    total: int
    page: int
    page_size: int


class GateApproveRequest(BaseModel):
    reviewer: str = "system"
    comment: Optional[str] = None


class GateRejectRequest(BaseModel):
    reviewer: str = "system"
    comment: str


class GateReviseRequest(BaseModel):
    """Phase 2.4: Request to revise and continue a gated stage."""
    reviewer: str = "system"
    comment: str
    revised_content: Optional[str] = None
